static/js/components/widgets/SelectField.tsx

```diff
@@ -74,29 +96,77 @@ export default function SelectField(props: Props): JSX.Element {
     selected = isNil(value) ? null : getSelectOption(value)
   }
 
+  const handleInputChanged = useCallback(
+    (input, reason) => {
+      if (reason.action === "input-blur") {
+        return
+      }
+      setSearchText(input)
+    },
+    [setSearchText]
+  )
+
+  const handleMenuClosed = useCallback(
+    () => setPlaceholder(searchText),
+    [setPlaceholder, searchText]
+  )
+
+  const handleMenuOpen = useCallback(
+    () => setSearchText(placeholder ?? ""),
+    [setSearchText, placeholder]
+  )
```

Comment:
`Select` hides the typed input element when the field loses focus.

This workaround replaces the placeholder with the typed search text so it is visible when the field loses focus. And populates the the input field when the field regains focus. When the field loses focus, the input, although hidden, is not persisted because of resulting glitches.